include/power/power.h

```diff
@@ -299,20 +322,27 @@ enum power_states pm_system_suspend(int32_t ticks);
 void _pm_power_state_exit_post_ops(enum power_states state);
 
 /**
- * @brief Application defined function for power state entry
+ * @brief Register a power management notifier
+ *
+ * Register the given notifier from the power management notification
+ * list.
  *
- * Application defined function for doing any target specific operations
- * for power state entry.
+ * @param notifier pm_notifier object to be registered.
  */
-void pm_notify_power_state_entry(enum power_states state);
+void pm_notifier_register(struct pm_notifier *notifier);
 
 /**
- * @brief Application defined function for sleep state exit
+ * @brief Unregister a power management notifier
+ *
+ * Remove the given notifier from the power management notification
+ * list. After that this object callbacks will not be called.
+ *
+ * @param notifier pm_notifier object to be unregistered.
  *
- * Application defined function for doing any target specific operations
- * for sleep state exit.
+ * @return true if the notifier was successfully removed, false
+ * otherwise.
  */
-void pm_notify_power_state_exit(enum power_states state);
+bool pm_notifier_unregister(struct pm_notifier *notifier);
```

Comment:
why bool? I would go for an int and use 0 for success and negative for any errors that might come out of it.

Reply:
also, I would go for pm_notifier_deregister

Reply:
This just return `false` in case the given notifier was never registered, otherwise always return `true`. I can change it EINVAL or ENOENT, though.

Reply:
unregister is far more used in our code base